blockchain/chain_sync/src/metrics.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use lazy_static::lazy_static;
+use prometheus::{
+    core::{AtomicU64, GenericCounter, GenericCounterVec, GenericGauge, Opts},
+    Histogram, HistogramOpts,
+};
+
+lazy_static! {
```

Comment:
Kudos for making this a singleton! One more suggestion, why don't we try once_cell: https://github.com/async-rs/async-std/issues/406 as people speak highly of it compared to lazy_static.

Reply:
Yeah, I looked into it, but the stdlib impl is only available in nightly and despite that, it's a container type that would require every routine that needs the `OnceCell<T>`  to call `get` to retrieve the inner `T`.

Reply:
I'm all ears if you understand the benefits we might get from adding that getter every time we collect a metric.

Reply:
From my understanding, the benefits would be:
- Lack of macros
- Faster hot get
I am not sure if that is enough for us to switch over at this point for this.

Reply:
Why is it faster than accessing a global static variable?

Is the concern with macros the incremental increase in compilation cost?

Reply:
oh, you don't need to use the OnceCell container. Instead you can use: https://docs.rs/once_cell/1.8.0/once_cell/#lazy-initialized-global-data. Anyways if you feel the refactor is quite complex, then lazy_static it is! As for perf improvements, we will have to measure to see the benefits in context to forest's access patterns.

Reply:
Not sure why its faster on hot get. That's just what someone said. But i dont think its significant. 
The concern with macros is increased compilation cost as well as the fact that there is code being generated. That being said, those are "meh" concerns for me... I'm ok with lazy_static. We use lazy_static in other parts of the codebase

Reply:
I don’t feel enticed, seems more fashion than function